changes needed for CDEPS PR #213

mvertens opened the pull request:
### Description of changes
changes needed for CDEPS PR #213

### Specific notes
Changes needed to enable flexible import to data ocean to be available in mediator history files

Contributors other than yourself, if any: None

CMEPS Issues Fixed (include github issue): #340 

Are changes expected to change answers? (specify if bfb, different at roundoff, more substantial) No - bfb

Any User Interface Changes (namelist or namelist defaults changes)? None

### Testing performed 
Verified bfb for docn test cases.
Verified that SMS_Ld5.f09_g17.B1850G.cheyenne_intel.allactive-cism-test_coupling and 
ERP_Ln9_Vnuopc.ne30_ne30_mg17.F2000climo.cheyenne_intel.cam-outfrq9s were bfb compared to cesm2_3_beta11

### Hashes used for testing: 
- [x] CESM:
  - cesm2_3_beta11

jedwards4b commented on mediator/med_phases_prep_ocn_mod.F90:
```diff
@@ -383,6 +382,13 @@ subroutine med_phases_prep_ocn_custom_cesm(gcomp, rc)
+       return
+    end if
+
+    call t_startf('MED:'//subname)
```
Shouldn't the t_startf be moved back to the top of the routine?

mvertens replied:
You need this where it is now - in case you need to return. Otherwise the model crashes if you try to return and there is no corresponding call to t_stopf.

mvertens commented:
@uturuncoglu @jedwards4b - I have updated the CDEPS documentation quite a bit. I have added new documentation for this PR as well as documenting the env_run.xml variables for each data component that appear in env_run.xml.
I think this PR is ready to be merged.

mvertens commented:
@uturuncoglu - can you look at the documentation and see if you are now comfortable with the PR.

mvertens commented:
@uturuncoglu - the CDEPS PR must be accompanied by this PR. Does it fit into your schedule to review and test it over the next day or two?

uturuncoglu commented:
@mvertens I am in ESMF metting. I could test it today with UFS.
